### PR TITLE
Clean up tests

### PR DESCRIPTION
**Pull request by atonks2**
This PR refactors all tests to use a common assertion library ([testify](https://github.com/stretchr/testify)). Some bugs were found in the tests and ICL records and corrected.

### CODE REVIEW

File: imageViewAnalysis.go [hunk 1]
```diff
@@ -573,7 +573,7 @@ func (ivAnalysis *ImageViewAnalysis) ConvenienceAmountUsabilityField() string {
 
 // AmountInWordsUsabilityField gets a string of the AmountInWordsUsability field
 func (ivAnalysis *ImageViewAnalysis) AmountInWordsUsabilityField() string {
-	return ivAnalysis.numericField(ivAnalysis.ConvenienceAmountUsability, 1)
+	return ivAnalysis.numericField(ivAnalysis.AmountInWordsUsability, 1)
```
Comment (atonks2) [inline imageViewAnalysis.go:576]: Looks like a copy/paste bug

File: returnDetailAddendumD.go [hunk 2]
```diff
@@ -184,6 +184,10 @@ func (rdAddendumD *ReturnDetailAddendumD) Validate() error {
 		return &FieldError{FieldName: "EndorsingBankRoutingNumber",
 			Value: rdAddendumD.EndorsingBankRoutingNumber, Msg: err.Error()}
 	}
+	if err := rdAddendumD.isNumeric(rdAddendumD.EndorsingBankItemSequenceNumber); err != nil {
```
Comment (atonks2) [inline returnDetailAddendumD.go:187]: There was a test that intended to ensure this validation was being performed, but the test was written such that no assertions were run if no error was returned.

This logic ensures Endorsing Bank Item Sequence Number is either numeric or empty. Technically this is a mandatory field, but in practice it's treated as optional. Other common libraries/tools will warn on empty sequence number, but it is not considered an error.

File: userPayeeEndorsement.go [hunk 3]
```diff
@@ -236,6 +236,10 @@ func (upe *UserPayeeEndorsement) Validate() error {
 		msg := fmt.Sprint(msgInvalid)
 		return &FieldError{FieldName: "UserRecordFormatType", Value: upe.UserRecordFormatType, Msg: msg}
 	}
+	if err := upe.isNumeric(upe.FormatTypeVersionLevel); err != nil {
```
Comment (atonks2) [inline userPayeeEndorsement.go:239]: There was a test that intended to ensure this validation was being performed, but the test was written such that no assertions were run if no error was returned.

**Review by adamdecaf — APPROVED** [review]
Thank you so so much! This is excellent cleanup.

**Pull request merged**